DataFormats/MuonReco/src/Muon.cc

```diff
@@ -144,23 +141,19 @@ unsigned int Muon::stationMask(ArbitrationType type) const {
   unsigned int totMask(0);
   unsigned int curMask(0);
 
-  for (std::vector<MuonChamberMatch>::const_iterator chamberMatch = muMatches_.begin();
-       chamberMatch != muMatches_.end();
-       chamberMatch++) {
+  for (auto& chamberMatch : muMatches_) {
     if (type == RPCHitAndTrackArbitration) {
-      if (chamberMatch->rpcMatches.empty())
+      if (chamberMatch.rpcMatches.empty())
         continue;
 
-      RPCDetId rollId = chamberMatch->id.rawId();
-      const int region = rollId.region();
-      int rpcIndex = 1;
-      if (region != 0)
-        rpcIndex = 2;
+      RPCDetId rollId = chamberMatch.id.rawId();
+      int rpcIndex = rollId.region() == 0 ? 1 : 2;
 
-      for (std::vector<MuonRPCHitMatch>::const_iterator rpcMatch = chamberMatch->rpcMatches.begin();
-           rpcMatch != chamberMatch->rpcMatches.end();
-           rpcMatch++) {
-        curMask = 1 << ((chamberMatch->station() - 1) + 4 * (rpcIndex - 1));
+      //    for (auto& rpcMatch : chamberMatch.rpcMatches) { // TO BE FIXED: there is clearly something odd here
```

Comment:
@jhgoh please confirm that what implemented here for the rpc matches (which faithfully reproduces what was coded so far in CMSSW) is actually the sought behaviour, and it did not reveal a bug in the the original code instead.
If it is confirmed being ok, I will remove the comment lines left as a warning.